### ISSUE TITLE
508-defect-1 ❗ Veteran rate tables have row headers missing the `scope="row"` attribute

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-cms#10372]
## Description

On https://www.va.gov/disability/compensation-rates/veteran-rates/, the Section 508 office recently received the following feedback:

> The rate tables accessed through the Job Aids Webpage are read by the JAWS software in rows. The rates are not identified as to the associated percentage or dependency scenario. This makes it difficult to go to the table and have a rate identified but not be able to associate a percentage or dependency scenario to it. At a minimum it would be best if we could label the rate box with the associated percentage. If the scenario could also be included, (Vet/Spouse/1parent/no kids info) that would offer additional clarity but we need the percentage at a minimum to know that the JAWS reader is identifying the proper column and data set

After reviewing the tables, I noticed that the row headers were missing the `scope="row"` attribute which is critical for screen reader users to navigate tables. Without this, we will fail to meet Section 508 compliance ([F91: Failure of Success Criterion 1.3.1 for not correctly marking up table headers](https://www.w3.org/TR/WCAG20-TECHS/F91.html)).

## Additional Information 
https://dsva.slack.com/archives/C52CL1PKQ/p1661181579663409?thread_ts=1660949215.680549&cid=C52CL1PKQ
![Screen Shot 2022-08-22 at 9 40 24 AM](https://user-images.githubusercontent.com/85581471/185973989-19d83bac-dd99-4439-a230-59c37011fc3b.png)

https://dsva.slack.com/archives/C5AGLBNRK/p1661176411346269?thread_ts=1660942058.542819&cid=C5AGLBNRK
Some discussion around usage of design system components.

## To reproduce:
*Please provide supporting artifacts as available.*

- 2022 Veterans disability compensation rates - https://www.va.gov/disability/compensation-rates/veteran-rates/
- Open any screen reading device
- Navigate to any of the tables
- Confirm that the row header is not announced

<img width="1300" alt="Screen Shot 2022-08-19 at 4 56 44 PM" src="https://user-images.githubusercontent.com/14154792/185706145-3d130261-6877-4c3a-9aaa-99df8e387b91.png">

## When does this work need to be done?
ASAP

## Tasks
- [ ] Read supporting slack threads linked in Additional Information
- [ ] Review notes re: Drupal tablefield module vs. design system component & summarize findings in comment
- [ ] Propose path forward, and if any ambiguity, review with Wes / Update tasks as needed
- [ ] TBD implementation details

## Acceptance criteria
- [ ]  row headers on  include the `scope="row"` attribute
- [ ] when these cells are updated to be table header cells to include the scope attribute, also make sure the role is updated as well (instead of cell it should be rowheader). The basic HTML would be:
`<th scope="row" role="rowheader"><th>`

**Comment by jilladams** [comment]
@RLHecht @laflannery just confirming you do/don't have anything to add here? If not, I'll unassign you and move to refinement.

**Comment by laflannery** [comment]
My only comment would be that when these cells are updated to be table header cells to include the scope attribute, also make sure the role is updated as well (instead of cell it should be rowheader).  The basic HTML would be:
`<th scope="row" role="rowheader"><th>`

**Comment by RLHecht** [comment]
@jilladams Thanks Jill, you can unassign me to this one!

**Comment by jilladams** [comment]
Thanks!

@wesrowe Labeled as both front end & Drupal eng for now, as we aren't sure the exact right way to implement this fix given the Drupal module vs. design component discussion in slack. Tasks / ACs updated to include researching that, coming to conclusion in order to work out path forward.

**Comment by allisonlu** [comment]
This will be a quick fix. I'll be updating the [Drupal table template](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/paragraphs/table.drupal.liquid) with the [design system Table component](https://design.va.gov/components/table), which should fix this issue.

https://github.com/department-of-veterans-affairs/content-build/blob/6b9027f7ec3109a83e7e7cc06c147f63c3c9db35/src/site/paragraphs/table.drupal.liquid#L51-L53

**Comment by jilladams** [comment]
Allison, note: Swirt is saying we can't use the component: https://dsva.slack.com/archives/C52CL1PKQ/p1661198335031119?thread_ts=1660949215.680549&cid=C52CL1PKQ

**Comment by jilladams** [comment]
Notes from call: 
- Has been a 508 issue since day 1 of CMS
- Will require Drupal updates: all tables generated by Drupal experience this issue. May be possible to resolve via upgrade to `tableview` module. This could be Sitewide CMS work, if PW doesn't have bandwidth. 
- The table in question needs both scopes.  It currently, correctly has scope="col" attributes on the top table header cells but it is missing table header cells (with scope="row") on the left most column of the table.  That is what we are trying to resolve
- Scope needs to include reviewing existing Tables, to verify what breaks / what works
- Follow up on FE: review whether or not we need any FE work, or Drupal work completes the task
- @swirtSJW to write Drupal ticket re: module upgrade

**Comment by EWashb** [comment]
Thanks @jilladams! @laflannery can help with any other CMS accessibility insight as well. If there are user experience concerns on the BE, please let me know how we can support yall. We have some backlog tickets for various updates to tables that live in #2932 and #8542.

**Comment by ryguyk** [comment]
I believe this will require changes in BOTH places discussed above:

1. Drupal will need to be updated to be able to capture the header information.
2. `content-build` template will need to be updated, as Allison pointed out. Unless I'm missing something, we _should_ be able to utilize the web component in the template (https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/paragraphs/table.drupal.liquid).

**Comment by allisonlu** [comment]
Going to sync with @chri5tia on this next week

**Comment by wesrowe** [comment]
@laflannery, can you please retest this in prod? If so, we fixed it via template and can remove the dependencies on 2 other issues.

**Comment by laflannery** [comment]
@wesrowe I am still seeing these as standard table cells, not table headers:
![image](https://user-images.githubusercontent.com/106678594/197020729-ac4863a3-9a9f-4d98-9044-a2678878b8fa.png)

**Comment by ryguyk** [comment]
The work on this got merged this morning. It should be live after today's deployment that is scheduled at 3PM ET.

**Comment by jilladams** [comment]
The earlier change is now deployed and this ticket can be re-checked in prod, as well.

**Comment by laflannery** [comment]
@jilladams and @wesrowe I checked prod this morning and confirmed all tables on the page now have column and row headers
